### PR TITLE
feat(frontend): support show create [materialized] view command

### LINKED ISSUE #6913: feat: support `show create [materialized] view`
### Is your feature request related to a problem? Please describe.

Allow users to print the SQL query used to create the (materialized) views.

```
show create [materialized] view <view_name>
```

Expects something like

```
         name         |                                             create_sql
----------------------+-----------------------------------------------------------------------------------------------------
 materialize.public.t | CREATE TABLE "materialize"."public"."t" ("a" "pg_catalog"."int4", "b" "pg_catalog"."text" NOT NULL)
```

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

See also:

- [SHOW CREATE MATERIALIZED VIEW | Materialize Documentation](https://materialize.com/docs/sql/show-create-materialized-view/)
- [SHOW CREATE VIEW | Materialize Documentation](https://materialize.com/docs/sql/show-create-view/)

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Support show create create [materialized] view commands.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

* SQL commands, functions, and operators
Support show create create [materialized] view commands:
```sql
dev=> create table t2(v1 int);
CREATE_TABLE
dev=> create materialized view mv1 as select * from t2;
CREATE_MATERIALIZED_VIEW
dev=> create view v1 as select * from t2;
CREATE_VIEW
dev=> show create materialized view public.mv1;
    Name    |                    Create Sql
------------+--------------------------------------------------
 public.mv1 | CREATE MATERIALIZED VIEW mv1 AS SELECT * FROM t2
(1 row)

dev=> show create view v1;
   Name    |             Create Sql
-----------+------------------------------------
 public.v1 | CREATE VIEW v1 AS SELECT * FROM t2
(1 row)
```

### Release note

Support show create create [materialized] view commands.

## Refer to a related PR or issue link (optional)
Resolve #6913

### CODE REVIEW

**Review by hzxa21 — APPROVED** [review]
LGTM.

Should we also support showing the view definition in a pg compatible way (via `pg_get_viewdef` or `pg_views`)?

**Comment by yezizp2012** [comment]
> LGTM.
> 
> Should we also support showing the view definition in a pg compatible way (via `pg_get_viewdef` or `pg_views`)?

Yes, we already support `definition` column in `pg_views`. 😄

**Review by fuyufjh — APPROVED** [review]
LGTM! Thank you a lot! ❤️

File: src/frontend/src/handler/show.rs [hunk 1]
```diff
@@ -131,6 +133,74 @@ pub fn handle_show_object(handler_args: HandlerArgs, command: ShowObject) -> Res
...
+        ShowCreateType::View => {
+            let view = schema
+                .get_view_by_name(&object_name)
+                .ok_or_else(|| RwError::from(CatalogError::NotFound("view", name.to_string())))?;
+
+            // We only stored original sql in catalog, uses parser to parse and format.
+            let stmt = Parser::parse_sql(&view.sql)
+                .map_err(|err| anyhow!("Failed to parse view create sql: {}, {}", view.sql, err))?;
+            assert!(stmt.len() == 1);
+            stmt[0].to_string()
```
Comment (BugenZhao) [inline src/frontend/src/handler/show.rs:173]: The SQL unparsed from the AST might be incorrect. :( https://github.com/risingwavelabs/risingwave/issues/6801 Will this be a problem, if this is for some programs to read?

Comment (BugenZhao) [reply src/frontend/src/handler/show.rs:173]: If so, I guess we need to carefully review the `fmt::Display` for all AST nodes and add a comprehensive test suite for it.

Comment (yezizp2012) [reply src/frontend/src/handler/show.rs:173]: 🥵 I think this is just for user to check how the mview and view are created.

> If so, I guess we need to carefully review the `fmt::Display` for all AST nodes and add a comprehensive test suite for it.

I think this's also needed, does the tests in sql_parser meet demand? It seems that we only have few tests there.

Comment (BugenZhao) [reply src/frontend/src/handler/show.rs:173]: I'm not sure. cc @lmatz

Comment (xxchan) [reply src/frontend/src/handler/show.rs:173]: We can test roundtrip parsing (`parse(unparse(parse(sql))) == parse(sql)` ) in sqlsmith

Comment (BugenZhao) [reply src/frontend/src/handler/show.rs:173]: Agree.

Comment (BugenZhao) [reply src/frontend/src/handler/show.rs:173]: > We can test roundtrip parsing (`parse(unparse(parse(sql))) == parse(sql)` ) in sqlsmith.

I'm afraid there's a case that some different SQLs are parsed to the same AST, which requires some binder context to distinguish. So when we unparse it, we cannot find the original representation, and this roundtrip test cannot cover this.

**Review by BugenZhao — APPROVED** [review]

File: src/frontend/src/handler/show.rs (hunk 1 above)
Comment (lmatz) [reply src/frontend/src/handler/show.rs:173]: I guess that's exactly why using AST instead of original sql to compare in xxchan's proposal?

**Pull request merged**